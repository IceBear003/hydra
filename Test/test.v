module find_first_one
(
    input [2047:0] data,
    output reg [10:0] cnt = 11'b0
);

reg update_trigger = 1'b0;

always @(data) begin
    
end

always @(data) begin
    
end

endmodule