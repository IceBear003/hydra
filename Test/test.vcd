$date
	Fri Apr 12 23:46:20 2024
$end
$version
	Icarus Verilog
$end
$timescale
	1ps
$end
$scope module tb_test $end
$var wire 4 ! position [3:0] $end
$var parameter 32 " PERIOD $end
$var reg 1 # clk $end
$var reg 8 $ data [7:0] $end
$scope module ffo $end
$var wire 1 # clk $end
$var wire 8 % data [7:0] $end
$var reg 4 & cnt [3:0] $end
$upscope $end
$upscope $end
$enddefinitions $end
$comment Show the parameter values. $end
$dumpall
b1010 "
$end
#0
$dumpvars
b0 &
b0 %
b0 $
1#
b0 !
$end
#5000
0#
#10000
b100 !
b100 &
1#
b11110000 $
b11110000 %
#15000
0#
#20000
b110 !
b110 &
1#
b11110110 $
b11110110 %
#25000
0#
#30000
b11 !
b11 &
1#
b11100000 $
b11100000 %
#35000
0#
#40000
b111 !
b111 &
1#
b10111111 $
b10111111 %
#45000
0#
#50000
1#
