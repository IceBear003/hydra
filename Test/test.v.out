#! /c/Source/iverilog-install/bin/vvp
:ivl_version "12.0 (devel)" "(s20150603-1539-g2693dd32b)";
:ivl_delay_selection "TYPICAL";
:vpi_time_precision + 0;
:vpi_module "C:\iverilog\lib\ivl\system.vpi";
:vpi_module "C:\iverilog\lib\ivl\vhdl_sys.vpi";
:vpi_module "C:\iverilog\lib\ivl\vhdl_textio.vpi";
:vpi_module "C:\iverilog\lib\ivl\v2005_math.vpi";
:vpi_module "C:\iverilog\lib\ivl\va_math.vpi";
S_000001fcf3d8bdb0 .scope module, "find_first_one" "find_first_one" 2 1;
 .timescale 0 0;
    .port_info 0 /INPUT 1 "clk";
    .port_info 1 /INPUT 4096 "data";
    .port_info 2 /OUTPUT 12 "position";
o000001fcf3dd6fa8 .functor BUFZ 1, C4<z>; HiZ drive
v000001fcf3da2e90_0 .net "clk", 0 0, o000001fcf3dd6fa8;  0 drivers
o000001fcf3dd6fd8 .functor BUFZ 4096, C4<zzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzz>; HiZ drive
v000001fcf3da2c30_0 .net "data", 4095 0, o000001fcf3dd6fd8;  0 drivers
v000001fcf3dd30a0_0 .var "i", 5 0;
v000001fcf3dd3140_0 .var "one_hot", 63 0;
v000001fcf3dd31e0_0 .var "position", 11 0;
v000001fcf3dd3280_0 .var "recursion_index", 5 0;
v000001fcf3dd3320_0 .var "update_trigger", 0 0;
E_000001fcf3d89ed0 .event posedge, v000001fcf3da2e90_0, v000001fcf3dd3320_0;
    .scope S_000001fcf3d8bdb0;
T_0 ;
    %pushi/vec4 0, 0, 12;
    %store/vec4 v000001fcf3dd31e0_0, 0, 12;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v000001fcf3dd3320_0, 0, 1;
    %pushi/vec4 0, 0, 6;
    %store/vec4 v000001fcf3dd3280_0, 0, 6;
    %pushi/vec4 0, 0, 64;
    %store/vec4 v000001fcf3dd3140_0, 0, 64;
    %end;
    .thread T_0;
    .scope S_000001fcf3d8bdb0;
T_1 ;
    %wait E_000001fcf3d89ed0;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v000001fcf3dd3320_0, 0, 1;
    %load/vec4 v000001fcf3da2c30_0;
    %load/vec4 v000001fcf3dd3280_0;
    %pad/u 12;
    %muli 64, 0, 12;
    %part/u 64;
    %cmpi/e 0, 0, 64;
    %jmp/0xz  T_1.0, 4;
    %load/vec4 v000001fcf3dd3280_0;
    %addi 1, 0, 6;
    %store/vec4 v000001fcf3dd3280_0, 0, 6;
    %pushi/vec4 1, 0, 1;
    %assign/vec4 v000001fcf3dd3320_0, 0;
    %jmp T_1.1;
T_1.0 ;
    %load/vec4 v000001fcf3dd3280_0;
    %ix/load 4, 6, 0;
    %flag_set/imm 4, 0;
    %store/vec4 v000001fcf3dd31e0_0, 4, 6;
    %load/vec4 v000001fcf3da2c30_0;
    %load/vec4 v000001fcf3dd3280_0;
    %pad/u 12;
    %muli 64, 0, 12;
    %part/u 64;
    %load/vec4 v000001fcf3da2c30_0;
    %load/vec4 v000001fcf3dd3280_0;
    %pad/u 12;
    %muli 64, 0, 12;
    %part/u 64;
    %subi 1, 0, 64;
    %inv;
    %and;
    %store/vec4 v000001fcf3dd3140_0, 0, 64;
    %pushi/vec4 0, 0, 6;
    %store/vec4 v000001fcf3dd30a0_0, 0, 6;
T_1.2 ;
    %load/vec4 v000001fcf3dd30a0_0;
    %pad/u 32;
    %cmpi/u 63, 0, 32;
    %jmp/0xz T_1.3, 5;
    %load/vec4 v000001fcf3dd3140_0;
    %load/vec4 v000001fcf3dd30a0_0;
    %part/u 1;
    %flag_set/vec4 8;
    %jmp/0xz  T_1.4, 8;
    %load/vec4 v000001fcf3dd30a0_0;
    %ix/load 4, 0, 0;
    %flag_set/imm 4, 0;
    %store/vec4 v000001fcf3dd31e0_0, 4, 6;
T_1.4 ;
    ; show_stmt_assign_vector: Get l-value for compressed += operand
    %load/vec4 v000001fcf3dd30a0_0;
    %pushi/vec4 1, 0, 6;
    %add;
    %store/vec4 v000001fcf3dd30a0_0, 0, 6;
    %jmp T_1.2;
T_1.3 ;
    %pushi/vec4 0, 0, 6;
    %store/vec4 v000001fcf3dd3280_0, 0, 6;
T_1.1 ;
    %jmp T_1;
    .thread T_1;
# The file index is used to find the file name in the following table.
:file_names 3;
    "N/A";
    "<interactive>";
    "test.v";
